werkzeug/routing.py

```diff
@@ -745,9 +745,9 @@ def _build_regex(rule):
         )
         self._regex = re.compile(regex, re.UNICODE)
 
-    def match(self, path):
+    def match(self, path, method=None):
         """Check if the rule matches a given path. Path is a string in the
-        form ``"subdomain|/path(method)"`` and is assembled by the map.  If
```

Comment:
I don't know why it was there, but it was not implemented. And changing the regex to work with this will be harder and more error prone than the method I'm suggesting I think.